network/src/sync/blocks/aggro.rs

```diff
@@ -0,0 +1,201 @@
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{sync::Arc, time::Duration};
+
+use crate::{Cache, Node, Payload, Peer, SyncBase, SyncInbound};
+use anyhow::*;
+use indexmap::IndexSet;
+use snarkos_storage::Digest;
+use snarkvm_dpc::BlockHeaderHash;
+use tokio::{
+    sync::{mpsc, RwLock},
+    time::Instant,
+};
+
+pub struct SyncAggro {
+    base: SyncBase,
+}
+
+struct BlockLocatorHashes {
+    hashes: Vec<BlockHeaderHash>,
+    last_update: Instant,
+}
+
+impl SyncAggro {
+    pub fn new(node: Node) -> (Self, mpsc::Sender<SyncInbound>) {
+        let (base, sender) = SyncBase::new(node);
+        let new = Self { base };
+        (new, sender)
+    }
+
+    async fn send_sync_messages(&mut self, nodes: Vec<(Peer, Vec<BlockHeaderHash>)>) -> Result<usize> {
+        info!("requested block information from {} peers", nodes.len());
+        let mut future_set = vec![];
+
+        for (peer, hashes) in nodes {
+            if let Some(handle) = self.base.node.peer_book.get_peer_handle(peer.address) {
+                future_set.push(async move {
+                    handle.send_payload(Payload::GetSync(hashes), None).await;
+                });
+            }
+        }
+        let sent = future_set.len();
+        futures::future::join_all(future_set).await;
+        Ok(sent)
+    }
+
+    pub async fn run(mut self) -> Result<()> {
+        let sync_nodes = self.base.find_sync_nodes().await?;
+
+        if sync_nodes.is_empty() {
+            return Ok(());
+        }
+
+        self.base.node.register_block_sync_attempt();
+
+        let block_locator_hashes = Arc::new(RwLock::new(BlockLocatorHashes {
+            hashes: SyncBase::block_locator_hashes(&self.base.node).await?,
+            last_update: Instant::now(),
+        }));
+
+        let peer_syncs = {
+            let block_locator_hashes = block_locator_hashes.read().await;
+            sync_nodes
+                .iter()
+                .map(|peer| (peer.clone(), block_locator_hashes.hashes.clone()))
+                .collect()
+        };
+
+        let hash_requests_sent = self.send_sync_messages(peer_syncs).await?;
+
+        if hash_requests_sent == 0 {
+            return Ok(());
+        }
+
+        let received_hashes = Arc::new(RwLock::new(Cache::<1024, 32>::default()));
```

Comment:
```suggestion
        let received_hashes = Arc::new(RwLock::new(Cache::<1024, 32>::default()));
```
it would be best to refer to `BLOCK_HEADER_SIZE` instead of `32` here

Reply:
it's block hash size, not header size. don't think we have a constant for that atm, just default digest size.